apps/mobile/src/queries/stacks/use-transactions-with-transfers.query.ts

```diff
@@ -0,0 +1,41 @@
+import { useCallback } from 'react';
+
+import { AddressTransactionWithTransfers } from '@stacks/stacks-blockchain-api-types';
+import { useQueries } from '@tanstack/react-query';
+
+import {
+  createGetAccountTransactionsWithTransfersQueryOptions,
+  useCurrentNetworkState,
+  useStacksClient,
+} from '@leather.io/query';
+
+export function useGetAccountTransactionsWithTransfersQueries(addresses: string[]) {
+  const network = useCurrentNetworkState();
+  const client = useStacksClient();
+  const queries = useQueries({
+    queries: addresses.map(address => ({
+      ...createGetAccountTransactionsWithTransfersQueryOptions({
```

Comment:
We will refactor away from this deprecated endpoint later?

Reply:
We will and I started looking into it but [it's a non-trivial task](https://github.com/leather-io/mono/pull/542#discussion_r1820903384). 

 In `createGetAccountTransactionsWithTransfersQueryOptions` we call `client.getAccountTransactionsWithTransfers` which uses `transactions_with_transfers` and [the API](https://docs.hiro.so/stacks/api/accounts/transactions-with-transfers) is deprecated now with no direct replacement provided. 
 
 I thought I had found a replacement but that was for a different API `transaction-with-transfers` where you need to provide the transaction. 
 
I think we need to use [Get address transactions](https://docs.hiro.so/stacks/api/transactions/address-transactions) and then get filter based on `"tx_type": "token_transfer",`.  I started some work on this in the larger [PR here](https://github.com/leather-io/mono/pull/542/files#r1822787679) but stopped as it was too much to change in one go. 

It's something I can add to https://linear.app/leather-io/issue/LEA-1695/increase-test-coverage-for-stx-balances and tackle it once we have more solid tests in place